packages/next-auth/src/lib/index.ts

```diff
@@ -59,6 +59,33 @@ export interface NextAuthConfig extends Omit<AuthConfig, "raw"> {
   }
 }
 
+function mergeCookiesAndSetCookiesHeaders(headers: Headers): string {
+  const setCookies = headers.get("set-cookie")
+  const cookies = headers.get("cookie");
+
+  if(!setCookies || (!cookies && !setCookies)) {
+    return cookies ?? "";
+  }
+
+  // We need to reparse the headers because the headers from Next returns [] on getSetCookie().
```

Comment:
If this is the case, it should be reported as a bug on Next.js.

Manually parsing cookies is error-prone, see below.